backend/open_webui/apps/webui/routers/auths.py

```diff
@@ -107,6 +121,90 @@ async def update_password(
+    LDAP_SERVER_PORT = request.app.state.config.LDAP_SERVER_PORT
+    LDAP_ATTRIBUTE_FOR_USERNAME = request.app.state.config.LDAP_ATTRIBUTE_FOR_USERNAME
+    LDAP_SEARCH_BASE = request.app.state.config.LDAP_SEARCH_BASE
+    LDAP_SEARCH_FILTERS = request.app.state.config.LDAP_SEARCH_FILTERS
+    LDAP_APP_DN = request.app.state.config.LDAP_APP_DN
+    LDAP_APP_PASSWORD = request.app.state.config.LDAP_APP_PASSWORD
+    USE_SSL = request.app.state.config.USE_SSL
+    CA_CERT_FILE = request.app.state.config.CA_CERT_FILE
+
+    if not ENABLE_LDAP:
+        raise HTTPException(400, detail="LDAP authentication is not enabled")
+    
+    try:
+        tls = Tls(validate=CERT_REQUIRED, version=PROTOCOL_TLS, ca_certs_file=CA_CERT_FILE, ciphers='ALL')
+    except Exception as e:
+        log.error(f"An error occurred on TSL: {str(e)}")
+        raise HTTPException(400, detail=str(e))
+    
+    try:
+        server = Server(host=LDAP_SERVER_HOST, port=LDAP_SERVER_PORT, get_info=ALL, use_ssl=USE_SSL, tls=tls)
+        connection_app = Connection(server, LDAP_APP_DN, LDAP_APP_PASSWORD, auto_bind='NONE', authentication='SIMPLE')
+        if not connection_app.bind():
+            raise HTTPException(400, detail="Application account bind failed")
+
+        search_success = connection_app.search(search_base=LDAP_SEARCH_BASE, search_filter=f'(&({LDAP_ATTRIBUTE_FOR_USERNAME}={form_data.user}){LDAP_SEARCH_FILTERS})', attributes=[f'{LDAP_ATTRIBUTE_FOR_USERNAME}', 'mail', 'cn'])
+        if not search_success:
+            raise HTTPException(400, detail="User not found in the LDAP server")
+        
+        entry = connection_app.entries[0]
+        username = str(entry[f'{LDAP_ATTRIBUTE_FOR_USERNAME}'])
+        mail = str(entry['mail'])
+        cn = str(entry['cn'])
+        user_dn = entry.entry_dn
+        if username == form_data.user:
+            connection_user = Connection(server, user_dn, form_data.password, auto_bind='NONE', authentication='SIMPLE')
+            if not connection_user.bind():
+                raise HTTPException(400, f"Authentication failed for {form_data.user}")
+            
+            user = Users.get_user_by_email(mail.lower())
+            if not user:
+                await signup(
+                    request,
+                    response,
+                    SignupForm(email=mail, password=str(form_data.password), name=cn)
+                )
+                user = Auths.authenticate_user(mail, password=str(form_data.password))
+
+            if user:
+                token = create_token(
+                    data={"id": user.id},
+                    expires_delta=parse_duration(request.app.state.config.JWT_EXPIRES_IN),
+                )
+
+                # Set the cookie token
+                response.set_cookie(
+                    key="token",
+                    value=token,
+                    httponly=True,  # Ensures the cookie is not accessible via JavaScript
```

Comment:
Ideally we should add `secure=True`  here, but I believe that requires `open-webui` to support HTTPS natively